src/main/java/org/candlepin/model/OwnerProduct.java

```diff
@@ -137,6 +140,7 @@ protected void applyObjectIds() {
         }
 
         this.ownerId = owner.getId();
+        this.productId = product.getId();
```

Comment:
We should add a check above to ensure the product ID is populated before we persist.